experimental/dds/tree2/docs/Shared Tree Philosophy.md

```diff
@@ -0,0 +1,199 @@
+
+Shared Tree's approach to doing all this while being extensible and maintainable relies on modularity and versioning, and is covered below.
+
+# A Design For Extensibility and Compatibility with Versioning
+
+Shared tree is a collaborative data-structure allowing cross version collaboration as well as support for loading all old formats.
+This imposes some very strict compatibility requirements that can make changing its implementation and functionality difficult.
+
+This puts a large tension between maintainability and extensibility.
+This leads to a design that is focused on optimizing for extensibility while minimizing the difficult of maintaining compatibility across these extensions.
+The approach Shared Tree takes for this is separation of concerns and version-ability of components.
+
+Concepts are split into three catagories based on compatibility requirements (see [SchemaVersioning](packages/dds/SchemaVersioning.md) for details of why these exist and what is in each):
+
+1. Critical for consistent behavior and thus must have compatibility forever to support old document and cross version collaboration.
+2. Only impacts the current instance of the application, and can be changed without maintaining identical behavior in previously supported cases.
+
+The Shared Tree design minimizes the amount of stuff (code, formats, data, etc) in the first category, as well as to keep the items in the first category as simple and independent as practical.
+
+For example, the merge resolution logic for sequences is defined as a single component, and if it needs changes that are incompatible (for example an improvement,ent to merge resolution), its possible to simply author a new version of this logic, and select between both at runtime based on either protocol version or schema.
+
+Additionally the Shared Tree architecture organizes these components such that if needed they can be replaced incrementally.
+For example, the tree reading and editing API is build on-top of cursors.
+A different version of this API can be authored, tested and adopted side by side with the old one with minimum difficulty.
+If the underlying tree storage (`forest`) changes, the tree API components (`editable-tree`) will not be impacted.
+Similarly if the cursor API needed to change, forest could add support for the new one (without dropping the old one), then users could migrate incrementally.
+This kind concurrent multi-version support is extra important for cases which fall into category one above (impact data compatibility not just API compatibility).
+For example shared tree can introduce a new version of editing primitives and/or merge semantics for a field kind (for example sequence, see below),
+which applications can opt into in their schema in a compatible way.
+
+Shared Tree is also designed to ensure applications using it can also adopt this same compatibility approach if needed while minimizing the frequency its required as well as the difficulty of doing it.
```

Comment:
```suggestion
Shared Tree is also designed to ensure applications using it can also adopt this same compatibility approach while minimizing the frequency and difficulty of doing so.
```